cl/api/webhooks.py

```diff
@@ -0,0 +1,244 @@
+            },
+        }
+        renderer = JSONRenderer()
+        json_bytes = renderer.render(
+            post_content,
+            accepted_media_type="application/json;",
+        )
+
+        webhook_event = WebhookEvent.objects.create(
+            webhook=webhook,
+            content=post_content,
+        )
+        send_webhook_event(webhook_event, json_bytes)
+
+
+def send_old_alerts_webhook_event(
+    webhook: Webhook, report: OldAlertReport
+) -> None:
+    """Send webhook event for old alerts
+
+    :param webhook:The Webhook object to send the event to.
+    :param report: A dict containing information about old alerts
+    :return None
+    """
+
+    serialized_very_old_alerts = []
+    serialized_disabled_alerts = []
+
+    for very_old_alert in report.very_old_alerts:
+        serialized_very_old_alerts.append(
+            DocketAlertSerializer(very_old_alert.da_alert).data
+        )
+
+    for disabled_alert in report.disabled_alerts:
+        serialized_disabled_alerts.append(
+            DocketAlertSerializer(disabled_alert.da_alert).data
+        )
+
+    post_content = {
+        "webhook": generate_webhook_key_content(webhook),
+        "payload": {
+            "old_alerts": serialized_very_old_alerts,
+            "disabled_alerts": serialized_disabled_alerts,
+        },
+    }
+    renderer = JSONRenderer()
+    json_bytes = renderer.render(
+        post_content,
+        accepted_media_type="application/json;",
+    )
+    webhook_event = WebhookEvent.objects.create(
+        webhook=webhook,
+        content=post_content,
+    )
+    send_webhook_event(webhook_event, json_bytes)
+
+
+def send_recap_fetch_webhook_event(
+    webhook: Webhook, fq: PacerFetchQueue
+) -> None:
+    """Send webhook event for processed PacerFetchQueue objects.
+
+    :param webhook: The Webhook object to send the event to.
+    :param fq: The PacerFetchQueue object related to the event.
+    :return None
+    """
+
+    payload = PacerFetchQueueSerializer(fq).data
+    post_content = {
+        "webhook": generate_webhook_key_content(webhook),
+        "payload": payload,
+    }
+    renderer = JSONRenderer()
+    json_bytes = renderer.render(
+        post_content,
+        accepted_media_type="application/json;",
+    )
+    webhook_event = WebhookEvent.objects.create(
+        webhook=webhook,
+        content=post_content,
+    )
+    send_webhook_event(webhook_event, json_bytes)
+
+
+def send_search_alert_webhook(
+    self,
+    results: SolrResponse,
+    webhook: Webhook,
+    search_type: str,
+    alert: Alert,
+) -> None:
+    """Send a search alert webhook event containing search results from a
+    search alert object.
+
+    :param self: The cl_send_alerts Command
+    :param results: The search results returned by SOLR for this alert.
+    :param webhook: The webhook endpoint object to send the event to.
+    :param search_type: The search type of the alert.
+    :param alert: The search alert object.
```

Comment:
Since we only use `self` and `search_type` to get the Solr URL, and since it's pretty weird to use `self` in this context, why not just pass the solr URL?

Reply:
Got it, changed, now only passing the Solr interface in order to reuse code in `cl_send_alerts`.